pkg/agent/validations_test.go

```diff
@@ -2,114 +2,271 @@ package agent
-				Formatter: new(logrus.TextFormatter),
-				Hooks:     make(logrus.LevelHooks),
-				Level:     logrus.DebugLevel,
-			}
-			hook := test.NewLocal(logger)
-
-			assert.Equal(t, tt.expectedResult, checkHostsValidations(cluster, logger))
-
-			assert.Equal(t, len(tt.expectedLogs), len(hook.Entries))
-			for _, expectedMsg := range tt.expectedLogs {
-
-				matchFound := false
-				for _, s := range hook.AllEntries() {
-					logLine, err := s.String()
-					assert.NoError(t, err)
-					if regexp.MustCompile(expectedMsg).Match([]byte(logLine)) {
-						matchFound = true
-					}
-				}
-				assert.True(t, matchFound, "Unable to find log trace for `%s`", expectedMsg)
-			}
+func TestLogValidationHistory(t *testing.T) {
+	tests := []struct {
+		name           string
+		inputHistory   *validationResultHistory
+		expectedResult string
+	}{
+		{
+			name: "success-first-seen",
+			inputHistory: &validationResultHistory{
+				numFailures:     0,
+				seen:            false,
+				currentStatus:   "success",
+				currentMessage:  "The validation succeeded",
+				previousStatus:  "",
+				previousMessage: "",
+			},
+			expectedResult: "debug",
+		},
+		{
+			name: "success-no-change",
+			inputHistory: &validationResultHistory{
+				numFailures:     0,
+				seen:            true,
+				currentStatus:   "success",
+				currentMessage:  "The validation succeeded",
+				previousStatus:  "success",
+				previousMessage: "The validation succeeded",
+			},
+			expectedResult: "",
+		},
+		{
+			name: "failure-first-seen",
+			inputHistory: &validationResultHistory{
+				numFailures:     0,
+				seen:            false,
+				currentStatus:   "failure",
+				currentMessage:  "The validation failed",
+				previousStatus:  "",
+				previousMessage: "",
+			},
+			expectedResult: "warning",
+		},
+		{
+			name: "failure-no-change",
+			inputHistory: &validationResultHistory{
+				numFailures:     1,
+				seen:            true,
+				currentStatus:   "failure",
+				currentMessage:  "The validation failed",
+				previousStatus:  "failure",
+				previousMessage: "The validation failed",
+			},
+			expectedResult: "",
+		},
+		{
+			name: "success-after-failure",
+			inputHistory: &validationResultHistory{
+				numFailures:     1,
+				seen:            true,
+				currentStatus:   "success",
+				currentMessage:  "The validation succeeded",
+				previousStatus:  "failure",
+				previousMessage: "The validation failed",
+			},
+			expectedResult: "info",
+		},
+		{
+			name: "failure-after-success",
+			inputHistory: &validationResultHistory{
+				numFailures:     1,
+				seen:            true,
+				currentStatus:   "failure",
+				currentMessage:  "The validation failed",
+				previousStatus:  "success",
+				previousMessage: "The validation succeeded",
+			},
+			expectedResult: "warning",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equal(t, tt.expectedResult, logValidationHistory(testLogPrefix, tt.inputHistory, logger))
```

Comment:
If we used the logrus hooks as in the previous test code, we could do `hook.LastEntry().Level` to get the level instead of having to return it as a string from the function for test purposes and hope they stay in sync.